firestore/angular-rewrite/src/app/restuarant-page/restuarant-page.component.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, OnInit, inject } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { Firestore, doc, docData, } from '@angular/fire/firestore';
+import { Restaurant } from '../restaurant-card/restaurant';
+
+@Component({
+  selector: 'app-restuarant-page',
+  templateUrl: './restuarant-page.component.html',
+  styleUrls: ['./restuarant-page.component.css']
+})
+export class RestuarantPageComponent implements OnInit {
+  restaurantID: string = "";
+  restaurantData: Observable<Restaurant> = new Observable();
+  private firestore: Firestore = inject(Firestore);
+
+  constructor(private route: ActivatedRoute) { }
+
+  ngOnInit(): void {
+    const id = this.route.snapshot.paramMap.get('id') as string;
+    this.restaurantID = id;
```

Comment:
Out of curiosity, what is the function of re-assigning the variable?

Reply:
Good catch! Originally this was included so that I could test whether or not I was passing the `restaurant ID` to `restaurant-page.component.html` and correctly fetching reviews in `review-list.component.ts`. Since the `id` is included in the `restaurantData` object, though, the `this.restaurantID` field isn't necessary.